main.js

```diff
@@ -160,4 +160,209 @@ const inventory = [
     originalStock: 10,
     sold: 8,
   },
-];
\ No newline at end of file
+];
+
+
+// Opdracht 1a:
+const tvType = inventory.map((inventoryTv) => {
+  return inventoryTv.type;
+});
+console.log(tvType);
+
+
+// Opdracht 1b:
+const tvSoldout = inventory.filter((inventorySoldout) => {
+  return inventorySoldout.originalStock === inventorySoldout.sold;
+})
+console.log(tvSoldout);
+
+
+// Opdracht 1c:
+// declareer een variabele waarin we de ambilight opslaan
+// loop over de array heen van tv's
+// check voor iedere vgl of die ambilight heeft
+
+const tvAmbilight = inventory.filter((inventoryAmbilight) => {
+  return inventoryAmbilight.options.ambiLight ;
+});
+console.log(tvAmbilight);
+
+
+// Opdracht 1d:
+inventory.sort((a, b) =>
+    a.price - b.price);
+console.log(inventory);
+
+
+//Opdracht 2a:
+const tvSold = inventory.map((inventorySold) => {
+  return inventorySold.sold;
+});
+console.log(tvSold);
+
+let counterTvSold = 0;
+for (let i = 0; i < tvSold.length; i++) {
+  counterTvSold += tvSold[i];
+};
+console.log(counterTvSold);
+
+
+//Opdracht 2b:
+const totalTvSold = document.getElementById('total-tv-sold');
+
+console.log(totalTvSold);
+
+totalTvSold.textContent = counterTvSold;
+
+
+//Opdracht 2c:
+const tvPurchased = inventory.map((inventoryPurchased) => {
+  return inventoryPurchased.originalStock;
+});
+console.log(tvPurchased);
+
+let counterTvPurchased = 0;
+for (let i = 0; i < tvPurchased.length; i++) {
+  counterTvPurchased += tvPurchased[i];
+};
+console.log(counterTvPurchased);
+
+
+//opdracht 2d:
+const totalTvPurchased = document.getElementById('total-tv-purchased');
+console.log(totalTvPurchased);
+
+totalTvPurchased.textContent = counterTvPurchased;
+
+
+//Opdracht 2e:
+const totalTvLeft = document.getElementById('total-tv-left');
+console.log(totalTvLeft);
+
+const tvLeft = counterTvPurchased - counterTvSold;
+console.log(tvLeft);
+
+totalTvLeft.textContent = tvLeft;
+// Hier vind hij echter dat type number not assignable is to type string?
+// terwijl dat eerder wel kon?
+
+
+//Opdracht 3a:
+const tvList = document.getElementById('tv-list');
+const listItem = inventory.map((tvItem) => {
+  return `<li>${tvItem.brand}</li>`;
+});
+tvList.innerHTML = `${listItem.join("")}`;
+
+
+//Opdracht 3b:
+function getTvList(tvObject) {
+  const tvList = document.getElementById('tv-list');
+  const listItem = tvObject.map((tvItem) => {
+    return `<li>${tvItem.brand}</li>`;
+  });
+  return tvList.innerHTML = `${listItem.join("")}`;
+};
+
+getTvList(inventory);
+
+
+//Opdracht 4a;
+// Ik krijg en hele array met tv's, ik weet niet goed hoe ik er 1 pecifieke uit moet krijgen dmv een functie
+function getOneTv(getTv) {
+  const oneTvString = getTv.map((oneTv) => {
+    return oneTv.brand + " " + oneTv.type + " - " + oneTv.name;
```

Comment:
Ik vind het volgende altijd wat makkelijker te lezen
```suggestion
    return `${oneTv.brand} ${oneTv.type} - ${oneTv.name}`;
```